ivy/__init__.py

```diff
@@ -59,6 +63,19 @@ def __new__(cls, dtype_str):
         return str.__new__(cls, dtype_str)
 
 
+class Shape(tuple):
+    def __new__(cls, shape_tup):
+        assert isinstance(shape_tup, (int, list, tuple, ivy.NativeShape))
+        if isinstance(shape_tup, int):
+            shape_tup = (shape_tup,)
+        elif isinstance(shape_tup, list):
+            shape_tup = tuple(shape_tup)
+        assert ivy.all(ivy.array([isinstance(v, int) for v in shape_tup]))
```

Comment:
no need to use `ivy.all` and `ivy.array` here, just `all([isinstance(v, int) for v in shape_tup])`

Reply:
some of the functions threw array type errors when I used them directly 😅

or let's say we merge it first and I'll find a fix in another pr since I'll be changing up the `shape_to_tuple` methods in a new PR after merging this?